modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts

```diff
@@ -275,12 +274,23 @@ const removeCaretFormat = function (editor: Editor, name: string, vars: FormatVa
 
     // TODO: Figure out how on earth this works, as it shouldn't since remove format
     //  definitely seems to require an actual Range
-    editor.formatter.remove(name, vars, expandedRng as Range);
+    editor.formatter.remove(name, vars, expandedRng as Range, similar);
     selection.moveToBookmark(bookmark);
   } else {
-    caretContainer = getParentCaretContainer(editor.getBody(), formatNode);
+    const caretContainer = getParentCaretContainer(editor.getBody(), formatNode);
     const newCaretContainer = createCaretContainer(false).dom();
-    const caretNode = insertFormatNodesIntoCaretContainer(parents, newCaretContainer);
+
+    // Find all formats present on the format node
+    const matchedFormats = FormatUtils.getFormatNamesPresent(editor, formatNode, [ 'removeformat', name ]);
+    // Filter out any matched formats that are 'visually' equivalent to the 'name' format since they are not unique formats on the node
+    const uniqueFormats = Arr.filter(matchedFormats, (fmtName) => !FormatUtils.areSimilarFormats(editor, fmtName, name));
```

Comment:
Can you explain a bit more about what a "unique" format is? Is this to cover the edge case if there are 2 formats with the same rules?

If so, maybe improve the comment about what `areSimilarFormats` does and why it's specifically matching on attributes, class, styles, etc....?